tests/integration-tests/tests/scaling/test_scaling.py

```diff
@@ -175,75 +181,90 @@ def test_scaling_special_cases(
-        target_capacity_reservation_arn=resource_group_arn, full_cluster_size=full_cluster_size
-    )
-    cluster = clusters_factory(cluster_config)
-
-    remote_command_executor = RemoteCommandExecutor(cluster)
-    scheduler_commands = scheduler_commands_factory(remote_command_executor)
-
-    upscale_cluster_config = pcluster_config_reader(
-        config_file="pcluster-upscale.config.yaml",
-        target_capacity_reservation_arn=resource_group_arn,
-        full_cluster_size=full_cluster_size,
-    )
 
-    # Scale up cluster
-    _assert_cluster_update_scaling(
-        cluster,
-        scheduler_commands,
-        region,
-        cluster_config=str(upscale_cluster_config),
-        expected_ec2_capacity=(0, full_cluster_size),
-        expected_compute_nodes=(0, full_cluster_size),
-        max_monitoring_time=minutes(max_scaling_time),
-        is_scale_down=False,
-    )
-    _assert_compute_nodes_in_cluster_are_from_odcr(cluster, region, resource_group_arn)
-    _assert_simple_job_succeeds(scheduler_commands, full_cluster_size, partition="q1")
-
-    # Apply custom partitions
-    output_file_path = render_jinja_template(
-        template_file_path=os.path.join(str(test_datadir), "include_custom_partition_large.sh.jinja"),
-        partition="q1",
-        cr="cr1",
-    )
-    remote_command_executor.run_remote_script(output_file_path, run_as_root=True)
-    _assert_simple_job_succeeds(scheduler_commands, full_cluster_size, partition="q1")
-
-    # Scale down cluster
-    downscale_cluster_config = pcluster_config_reader(
-        config_file="pcluster-downscale.config.yaml",
-        target_capacity_reservation_arn=resource_group_arn,
-        downscaled_cluster_size=downscaled_cluster_size,
-        full_cluster_size=full_cluster_size,
-    )
-    _assert_cluster_update_scaling(
-        cluster,
-        scheduler_commands,
-        region,
-        cluster_config=str(downscale_cluster_config),
-        expected_ec2_capacity=(downscaled_cluster_size, full_cluster_size),
-        expected_compute_nodes=(downscaled_cluster_size, full_cluster_size),
-        max_monitoring_time=minutes(max_scaling_time),
-        is_scale_down=True,
-    )
-    _assert_simple_job_succeeds(scheduler_commands, downscaled_cluster_size, partition="q1")
-
-    # Scale up the cluster
-    _assert_cluster_update_scaling(
-        cluster,
-        scheduler_commands,
-        region,
-        cluster_config=str(upscale_cluster_config),
-        expected_ec2_capacity=(downscaled_cluster_size, full_cluster_size),
-        expected_compute_nodes=(downscaled_cluster_size, full_cluster_size),
-        max_monitoring_time=minutes(max_scaling_time),
-        is_scale_down=False,
-    )
-    _assert_compute_nodes_in_cluster_are_from_odcr(cluster, region, resource_group_arn)
-    _assert_simple_job_succeeds(scheduler_commands, full_cluster_size, partition="q1")
+    for launch_api in ["run_instances", "create_fleet"]:
```

Comment:
Since we are going to create a new cluster anyway if we parameterize the test maybe we can speed up the test. Not sure if they may be executed in parallel

Reply:
We want to reuse the ODCR. Using parametrize would create the ODCR stack again.
Also, this test deploys a lot of resources and so parallelism may come at the risk of hitting account quota limits easily.